expr/interval_compound_literal.go

```diff
@@ -0,0 +1,154 @@
+package expr
+
+import (
+	"errors"
+
+	"github.com/substrait-io/substrait-go/proto"
+	"github.com/substrait-io/substrait-go/types"
+)
+
+type intervalDateParts struct {
```

Comment:
I think there is too much composition and cuteness going on in this patch.

Let's just write a simple struct with public fields that implements Literal interface. We shouldn't use protoliteral. We shouldn't use private structs inside of something else.

Sometimes the simplest thing is the best. Let's keep it simple.